### ISSUE TITLE
Distillation of large ViT

### ISSUE THREAD

**Issue by SimJeg** [baaivision/EVA#25]
Hello,

In your opinion, what is the best way to distill a large vision transformer (eg. ViT-g) into a small one (eg. ViT-B) ?

Seems there are many alternatives: MIM as for EVA, distillation token as in DeiT, more classical techniques such as response based or feature based methods etc.

Thanks,
Simon

**Comment by Yuxin-CV** [comment]
Hi @SimJeg, thanks for your interest in EVA.

In my opinion, the best way to pre-train a vanilla ViT vision encoder currently, is to reconstruct the masked-out image-text aligned vision features conditioned on visible image patches, _i.e._, to perform MIM with CLIP vision features as reconstruction targets.

**Comment by Yuxin-CV** [comment]
Here is a detailed recent study on this topic: https://arxiv.org/pdf/2210.10615.pdf

<img width="1177" alt="image" src="https://user-images.githubusercontent.com/57927171/208300483-ba505192-1235-41fe-883c-7e007106dacc.png">

**Comment by Yuxin-CV** [comment]
We have performed MIM pre-training on vanilla ViT-L encoder (303M) with features EVA-CLIP vision branch (1.0B) as targets. The fine-tuned ViT-L encoder can achieve up to 89.2 top-1 acc on IN-1K: https://github.com/baaivision/EVA/tree/master/eva#eva-l-learning-better-mim-representations-from-eva-clip

**Comment by Yuxin-CV** [comment]
I think it depends on your budget as well as the size of your downstream dataset of interest. If you have enough compute / time / budgets, it is better to train on a larger dataset for a general representation. If you only have limited resources, and your your downstream dataset of interest is not very small (e.g., > 0.1M). I suggest directly pre-train on that dataset.

For reference, here are two papers about MIM pre-training on the smaller scale MS-COCO dataset, which only has 0.1M images.
- Are Large-scale Datasets Necessary for Self-Supervised Pre-training? (https://arxiv.org/abs/2112.10740)
- Exploring Long-Sequence Masked Autoencoders (https://arxiv.org/abs/2210.07224)

**Comment by SimJeg** [comment]
Thanks, you've been very helpful. Feel free to close the issue
Hope you can distill a ViT-B which is more suited for deployment

**Comment by Yuxin-CV** [comment]
We have distilled some ViT-B, we will release them soon. Please stay tuned.

**Comment by SimJeg** [comment]
@Yuxin-CV , an interesting read on how to perform well "classical" distillation that is not mentioned in the EVA paper or this thread: [Knowledge distillation: A good teacher is patient and consistent](https://arxiv.org/abs/2106.05237). 

Very good news for the planned ViT-B release, it will be very useful :)

**Comment by vateye** [comment]
Hi, do you have any plans to release the distilled ViT-B recently?

**Comment by SimJeg** [comment]
Hi @Yuxin-CV , any news on a release of a ViT-B distilled model ?

**Comment by Yuxin-CV** [comment]
Hi @vateye @SimJeg, we will release ViT-B at the end of March

**Comment by chaoyan1037** [comment]
> We have performed MIM pre-training on vanilla ViT-L encoder (303M) with features EVA-CLIP vision branch (1.0B) as targets. The fine-tuned ViT-L encoder can achieve up to 89.2 top-1 acc on IN-1K: https://github.com/baaivision/EVA/tree/master/eva#eva-l-learning-better-mim-representations-from-eva-clip

Hi @Yuxin-CV, thanks for sharing your results, it looks great! In your paper, you pretrain a larger model (vanilla ViT-g) with features from a smaller model (ViT-L CLIP (303M)). Just curious if you have tried use EVA to pretrain vanilla ViT-g or ViT-L encoder with features from ViT-g CLIP vision branch (1.0B)?

**Comment by Yuxin-CV** [comment]
> > We have performed MIM pre-training on vanilla ViT-L encoder (303M) with features EVA-CLIP vision branch (1.0B) as targets. The fine-tuned ViT-L encoder can achieve up to 89.2 top-1 acc on IN-1K: https://github.com/baaivision/EVA/tree/master/eva#eva-l-learning-better-mim-representations-from-eva-clip
> 
> Hi @Yuxin-CV, thanks for sharing your results, it looks great! In your paper, you pretrain a larger model (vanilla ViT-g) with features from a smaller model (ViT-L CLIP (303M)). Just curious if you have tried use EVA to pretrain vanilla ViT-g or ViT-L encoder with features from ViT-g CLIP vision branch (1.0B)?

Hi @chaoyan1037, we have tried to pre-train ViTs via reconstructing EVA-CLIP-g vision features. The results are promising. We plan to release them to the community before the end of March.

**Comment by chaoyan1037** [comment]
> > > We have performed MIM pre-training on vanilla ViT-L encoder (303M) with features EVA-CLIP vision branch (1.0B) as targets. The fine-tuned ViT-L encoder can achieve up to 89.2 top-1 acc on IN-1K: https://github.com/baaivision/EVA/tree/master/eva#eva-l-learning-better-mim-representations-from-eva-clip
> > 
> > 
> > Hi @Yuxin-CV, thanks for sharing your results, it looks great! In your paper, you pretrain a larger model (vanilla ViT-g) with features from a smaller model (ViT-L CLIP (303M)). Just curious if you have tried use EVA to pretrain vanilla ViT-g or ViT-L encoder with features from ViT-g CLIP vision branch (1.0B)?
> 
> Hi @chaoyan1037, we have tried to pre-train ViTs via reconstructing EVA-CLIP-g vision features. The results are promising. We plan to release them to the community before the end of March.

Thanks for the info! Look forward to your progress!

**Comment by Yuxin-CV** [comment]
@vateye @chaoyan1037 @SimJeg 
For smaller models, please refer to https://github.com/baaivision/EVA/tree/master/EVA-02

**Comment by SimJeg** [comment]
@Yuxin-CV amazing work, congrats !!!
